# (PART) MOTHUR BIOINFORMATICS {-}

# Mothur Bioinformatics: Getting Started with the Pipeline

## Tutorial References

Before diving into the Mothur bioinformatics pipeline, make sure to familiarize yourself with the following essential references:

1. **Snakemake Workflow:** Explore the comprehensive [Snakemake workflow](https://github.com/wclose/mothurPipeline/tree/master) designed to run Mothur within a conda environment.

2. **Full Mothur Tutorial:** Refer to the official [Mothur tutorial](https://mothur.org/wiki/miseq_sop/) for a detailed guide on utilizing Mothur in bioinformatics analysis.

# Setting Up Mothur in a Conda Environment

To streamline the Mothur pipeline, we've extracted and simplified key steps from the detailed tutorial available [here](https://github.com/wclose/mothurPipeline/tree/master). Follow these steps for a smooth start:

1. **Clone the Snakemake Workflow:**
   - Clone the Snakemake workflow repository to your local machine:
     ```bash
     git clone https://github.com/wclose/mothurPipeline.git
     ```

2. **Navigate to the Workflow Directory:**
   - Change into the cloned directory:
     ```bash
     cd mothurPipeline
     ```

3. **Set Up Conda Environment:**
   - Create a conda environment for Mothur:
     ```bash
     conda env create -f environment.yaml
     ```

4. **Activate Conda Environment:**
   - Activate the conda environment:
     ```bash
     conda activate mothur
     ```

5. **Run the Snakemake Workflow:**
   - Execute the Snakemake workflow to start the Mothur analysis:
     ```bash
     snakemake --use-conda
     ```

These steps provide a simplified entry point to using Mothur within a conda environment. For detailed explanations and additional features, refer to the [full Snakemake workflow](https://github.com/wclose/mothurPipeline/tree/master) repository.

## Create a `mothur` YAML file 

```bash
name: mothur48
channels:
    - conda-forge
    - bioconda
    - defaults
dependencies:
    - mothur =1.48.0
    - vsearch =2.22.1
```

## Create `mothur` environment using YAML File
```bash
conda activate base
conda env create -n mothur48 --file mothur48.yml
conda activate mothur48 
```

## Download References Databases
- Download Silva alignment reference database.
- Optionally create Silva classifier from Silva alignments.
- Download RDP classifier.
- Get any other suitable classifier compatible with Mothur.

The download script is in the repo, saved in the workflow/scripts directory. Run the following on command line or use the snakemake rules.

```bash
bash workflow/scripts/mothurReferences.sh
```

<br>

# (PART) MOTHUR CLASSIFICATION METHODS {-}
# Overview of Mothur classification methods
There are four methods that can be used to profile microbial communities present in a sample. Here we briefly decribe each method:

## Classify OTUs
- OTUs (Operational Taxonomic Units (OTUs)) are clusters of similar sequences and are commonly accepted as analytical units in microbial profiling when using 16S rRNA gene markers.

## Classify Phylotypes
- A phylotype in microbiome research is a DNA sequence or group of sequences sharing more than an arbitrarily chosen level of similarity of a 16S rRNA gene marker.

## Classify ASVs
- ASVs Amplicon Sequence Variants (ASVs)in microbiome research is any inferred single DNA sequences recovered from a bioinformatics analysis of 16S rRNA marker genes.
- ASV is typically really a cluster of sequences that are one or two bases apart from each other.

## Classify Phylogenies
- Microbial phylogenies are from gene sequence homologies.  Models of mutation determine the most-likely evolutionary histories.

<br>

# Preliminary OTU analysis using `Mothur`
The preliminary analysis (`alpha_beta_diversity rule`) is part of the bioinformatics analysis. It includes:

- Creating reads count for each group.
- Subsampling for downstream analysis.
- Rarefaction.
- Computing Alpha diversity metrics.
- Computing Beta diversity metrics.
- Getting sample distances.
- Constructing sample phylip tree.
- Generating ordination matrices including PCoA and NMDS.
- More as needed...
